apps/passport-server/src/services/telegramService.ts

```diff
@@ -0,0 +1,125 @@
+import { EdDSATicketPCD, EdDSATicketPCDPackage } from "@pcd/eddsa-ticket-pcd";
+import { Bot, InlineKeyboard } from "grammy";
+import { fetchTelegramConversation } from "../database/queries/telegram/fetchTelegramConversation";
+import { insertTelegramConversation } from "../database/queries/telegram/insertTelegramConversation";
+import { markTelegramUserAsVerified } from "../database/queries/telegram/updateTelegramConversation";
+import { ApplicationContext } from "../types";
+import { logger } from "../util/logger";
+
+export class TelegramService {
+  private context: ApplicationContext;
+  private bot: Bot;
+  private privateChatId: string;
+
+  public constructor(
+    context: ApplicationContext,
+    bot: Bot,
+    privateChatId: string
+  ) {
+    this.context = context;
+    this.bot = bot;
+    this.privateChatId = privateChatId;
+
+    this.bot.command("start", async (ctx) => {
```

Comment:
an issue i've run into with the bot is that it seems like it's only possible to have one bot instance per api key running. this means that during deploy, while the new server is spinning up and the previous one is still up, there will be a brief time during which we're going to have two instance running, which will cause the 2nd one to throw an exception and fail to connect. one hacky way to resolve this is to wait a minute before starting the bot on server startup

Reply:
I've added a configurable delay for bot startup